src/graphql/data/nft/Asset.ts

```diff
@@ -117,74 +121,100 @@ export function useAssetsQuery(
   last?: number,
   before?: string
 ) {
-  const queryData = useLazyLoadQuery<AssetQuery>(assetQuery, {
-    address,
-    orderBy,
-    asc,
-    filter,
-    first,
-    after,
-    last,
-    before,
-  })
+  const vars = useMemo(
+    () => ({ address, orderBy, asc, filter, first, after, last, before }),
+    [address, after, asc, before, filter, first, last, orderBy]
+  )
+  const [queryOptions, setQueryOptions] = useState({ fetchKey: 0 })
+  const queryData = useLazyLoadQuery<AssetQuery>(assetQuery, vars, queryOptions)
+
   const { data, hasNext, loadNext, isLoadingNext } = usePaginationFragment<AssetPaginationQuery, any>(
     assetPaginationQuery,
     queryData
   )
 
-  const assets: GenieAsset[] = data.nftAssets?.edges?.map((queryAsset: { node: any }) => {
-    const asset = queryAsset.node
-    const ethPrice = parseEther(
-      asset.listings?.edges[0]?.node.price.value?.toLocaleString('fullwide', { useGrouping: false }) ?? '0'
-    ).toString()
-    return {
-      id: asset.id,
-      address: asset.collection.nftContracts[0]?.address,
-      notForSale: asset.listings?.edges.length === 0,
-      collectionName: asset.collection?.name,
-      collectionSymbol: asset.collection?.image?.url,
-      imageUrl: asset.image?.url,
-      animationUrl: asset.animationUrl,
-      marketplace: asset.listings?.edges[0]?.node.marketplace.toLowerCase(),
-      name: asset.name,
-      priceInfo: asset.listings
-        ? {
-            ETHPrice: ethPrice,
-            baseAsset: 'ETH',
-            baseDecimals: '18',
-            basePrice: ethPrice,
-          }
-        : undefined,
-      susFlag: asset.suspiciousFlag,
-      sellorders: asset.listings?.edges.map((listingNode: { node: SellOrder }) => {
+  // Poll for updates.
+  const POLLING_INTERVAL = ms`999ms`
+  const environment = useRelayEnvironment()
+  const refresh = useCallback(async () => {
+    const length = data.nftAssets?.edges?.length
+    // Only poll if the user has not started paging.
+    // Polling after paging requires reloading *all* pages, which is untenable.
+    // TODO(cbachmeier): Refactor CollectionNfts to pass the first visible cursor to useAssetsQuery,
+    // to allow for polling of the visible page only. This would make polling tenable after the user has paged.
+    if (length > DEFAULT_ASSET_QUERY_AMOUNT) return
```

Comment:
Do assets still refresh after the `queryCacheExpirationTime` expires?

Reply:
AFAICT they'll only refresh if they are re-rendered, because that will trigger a re-fetch. They do not refresh when the time expires, nor do they do that right now.